### DISCUSSION TITLE
Is it possible to get all bounding boxes and their associated text

### DISCUSSION THREAD

**Discussion by shahafabileah** [ocrmypdf/OCRmyPDF#1319] [Q&A]
Hey everyone,
I'd like to OCR a PDF document and then get structured info about all the bounding boxes in the document and the relevant text fragments in each one.  Is that possible?
Looking at the source code, I see that there's a PdfContext that is instantiated before the pipeline runs.  It appears that it should be possible to do something like context.get_page_contexts, and with each do page_context.pageinfo.get_textareas (or something like that).  But, this context is not returned when running the pipeline.  Is there another way to get it?

```
def _run_pipeline(
    options: argparse.Namespace,
    plugin_manager: OcrmypdfPluginManager,
) -> ExitCode:
    with (...):
        ...
        context = PdfContext(options, work_folder, origin_pdf, pdfinfo, plugin_manager)

        # Validate options are okay for this pdf
        validate_pdfinfo_options(context)

        # Execute the pipeline
        optimize_messages = exec_concurrent(context, executor)

        exitcode = report_output_pdf(options, start_input_file, optimize_messages)
        return exitcode
```

For my reference:
https://github.com/ocrmypdf/OCRmyPDF/blob/cb2f090c60808fd130a874f74d8cb44aa344d477/src/ocrmypdf/_pipelines/ocr.py#L184
https://github.com/ocrmypdf/OCRmyPDF/blob/cb2f090c60808fd130a874f74d8cb44aa344d477/src/ocrmypdf/_jobcontext.py#L63
https://github.com/ocrmypdf/OCRmyPDF/blob/cb2f090c60808fd130a874f74d8cb44aa344d477/src/ocrmypdf/pdfinfo/info.py#L1004

**Comment by jbarlow83** [comment]
Those APIs are more geared toward OCRmyPDF's internal use.

The easiest way to get bounding boxes would be to first run OCRmyPDF on any PDFs you have that are missing text, and then using a library like pdfminer.six or pdfplumber (which provides a higher level interface for the former). These library detect text in the PDF, whether it was produced by OCR or whether the PDF initially had text.
